tests/test_download.py

```diff
@@ -76,89 +89,110 @@ def test_download_from_s3_with_endpoint_URL(self, remote_local_file: Any):
             mock_default.return_value = credentials_mock, None
-            download_from_gcs(out, tmp.name)
+            downloader = GCSDownloader()
+            downloader.download(out, tmp)
             mock_client.assert_called_once_with(credentials=credentials_mock)
-            assert os.path.isfile(tmp.name)
+            assert os.path.isfile(tmp)
 
     @pytest.mark.usefixtures('gcs_hmac_client', 'gcs_test', 'remote_local_file')
     def test_filenotfound_exception(self, remote_local_file: Any):
         with pytest.raises(FileNotFoundError):
             mock_remote_filepath, mock_local_filepath = remote_local_file(cloud_prefix='gs://')
-            download_from_gcs(mock_remote_filepath, mock_local_filepath)
+            downloader = GCSDownloader()
+            downloader.download(mock_remote_filepath, mock_local_filepath)
 
     @pytest.mark.usefixtures('gcs_hmac_client', 'gcs_test', 'remote_local_file')
     def test_invalid_cloud_prefix(self, remote_local_file: Any):
         with pytest.raises(ValueError):
             mock_remote_filepath, mock_local_filepath = remote_local_file(cloud_prefix='s3://')
-            download_from_gcs(mock_remote_filepath, mock_local_filepath)
+            downloader = GCSDownloader()
+            downloader.download(mock_remote_filepath, mock_local_filepath)
 
     def test_no_credentials_error(self, remote_local_file: Any):
         """Ensure we raise a value error correctly if we have no credentials available."""
         with pytest.raises(ValueError):
             mock_remote_filepath, mock_local_filepath = remote_local_file(cloud_prefix='gs://')
-            download_from_gcs(mock_remote_filepath, mock_local_filepath)
+            downloader = GCSDownloader()
+            downloader.download(mock_remote_filepath, mock_local_filepath)
 
 
 class TestDatabricksUnityCatalog:
 
-    def test_invalid_prefix_from_db_uc(self, remote_local_file: Any):
-        with tempfile.NamedTemporaryFile(delete=True, suffix='.txt') as tmp:
-            file_name = tmp.name.split(os.sep)[-1]
-            mock_remote_filepath, _ = remote_local_file(cloud_prefix='dbfs:/Volumess',
+    @pytest.mark.parametrize('cloud_prefix', ['dbfs:/Volumess', 'dbfs:/Content'])
+    def test_invalid_prefix_from_db_uc(self, remote_local_file: Any, cloud_prefix: str):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            file_name = os.path.join(tmp_dir, 'file.txt')
+            mock_remote_filepath, _ = remote_local_file(cloud_prefix=cloud_prefix,
                                                         filename=file_name)
-            with pytest.raises(Exception, match='Expected path prefix to be.*'):
-                download_from_databricks_unity_catalog(mock_remote_filepath, tmp.name)
+            with pytest.raises(Exception, match='Expected path prefix to be `dbfs:/Volumes`.*'):
+                downloader = DatabricksUnityCatalogDownloader()
+                downloader.download(mock_remote_filepath, file_name)
```

Comment:
nit. Can you add one more test to cover the case where DatabricksError is raised.

Reply:
done